### ISSUE TITLE
Tab layout not working when tab name is unicode

### ISSUE THREAD

**Issue by spapas** [maraujop/django-crispy-forms#396]
If you create a ``Tab`` whose name has unicode (for example greek) characters then you'll get an empty css_id.

This problem is created in the line ```self.css_id = slugify(self.name)``` of ``crispy_forms.bootstrap.py`` because of the unfortunate way of how ``django.template.defaultfilters.slugify`` works (only slugifies ascii characters).

This is a very common problem, I've experienced it in almost every project whose creators are located in countries whose alphabet is based on ascii (latin) charadcters (since they won't use unicode characters). 

In any  case, the solution is simple: Change line 
``self.css_id = slugify(self.name)`` to ``self.css_id = slugify(unidecode.unidecode(name))`` (and don't forget to ``import unidecode``). ``unidecode`` is a great library that translates all non-unicode characters to their latin equivalents, for example it will translate ``γεια σας`` to ``geia sas`` so then ``slugify`` will correctly work. 

However, unidecode actually has a GPL license (!) so I won't provide a patch since I am not really sure that we could use it (please see similar discussion in https://github.com/alex/django-taggit/pull/273). 

A different solution for that is to add a ``css_id`` kwarg to ``Tab`` so that people can pass the name of the css they want themselves (I won't provide it either since I am using a different solution). 

Finally, people that quickly want their unicode-named tabs to work, please try the following (that's what I do) :

```
from crispy_forms.bootstrap import Tab as NoUnicodeTab

class Tab(NoUnicodeTab):
    def __init__(self, name, *fields, **kwargs):
        super(Tab, self).__init__(name, *fields, **kwargs)
        self.css_id = slugify(unidecode.unidecode(name))
```

Then ``Tab('επιτέλους', 'field')`` *will work* !

**Comment by carltongibson** [comment]
This is a bug. But it's not one I think we can take on right now. 

Ideally we need a fix in Django, but that's not going to happen: 

> The drawbacks of implementing something more complicated outweigh the advantages at this stage.
> [#8391 (Slugify Template Filter Poorly Encodes Non-English Strings) – Django](https://code.djangoproject.com/ticket/8391#comment:38)

I'm going to take "Override `__init__`" as per your example as the correct answer for now. 

I'm not happy to add a `css_id` keyword to `Tab` here, as that opens the door to a long line of equivalent changes elsewhere. 

I **would consider** a documented, tested pull request that solved this in a simple and general manner. I am happy to discuss ideas for that (here, say) before anyone takes on coding. 

I'm inclined to agree with the comment on the Django ticket above.

**Comment by carltongibson** [comment]
As a follow up, it looks like this is [coming in Django 1.9](https://github.com/django/django/commit/f8cc464452f495fce2a3d6f7494396c8f798a1e6#diff-f614fc7a7dbb2882be1692a448abd21fR413)

**Comment by spapas** [comment]
Hello, sorry for returning to this issue after almost two years but it bit me once again and I lost a lot of time researching it (because I'd forgotten completely about it). In any case, I've tried using Tabs with django-crispy-forms 1.6.0 and django 1.10 and still no luck :( 

So, after some research I had to resort to the solution I posted above (https://github.com/maraujop/django-crispy-forms/issues/396#issue-54419370).

However, I'd like to mention that for django-taggit, the solution that was selected was to use the unidecode library iff found without explicitly requiring it in setup.py (check my comment on https://github.com/alex/django-taggit/pull/273#issuecomment-66955534 and PR alex/django-taggit#315). 

Couldn't that be also used here? I'd be happy to provide a small PR.
